stackslib/src/config/mod.rs

```diff
@@ -1996,134 +2008,151 @@ impl BurnchainConfigFile {
+    /// ---
+    /// @default: `/tmp/stacks-node-{current_timestamp}`
+    /// @notes:
+    ///   - For persistent mainnet or testnet nodes, this path must be explicitly configured to
+    ///     a non-temporary location.
     pub working_dir: String,
     /// The IPv4 address and port (e.g., "0.0.0.0:20443") on which the node's HTTP RPC server
     /// should bind and listen for incoming API requests.
-    ///
-    /// Default: `"0.0.0.0:20443"`
+    /// ---
+    /// @default: `"0.0.0.0:20443"`
     pub rpc_bind: String,
     /// The IPv4 address and port (e.g., "0.0.0.0:20444") on which the node's P2P networking
     /// service should bind and listen for incoming connections from other peers.
-    ///
-    /// Default: `"0.0.0.0:20444"`
+    /// ---
+    /// @default: `"0.0.0.0:20444"`
     pub p2p_bind: String,
     /// The publicly accessible URL that this node advertises to peers during the P2P handshake
     /// as its HTTP RPC endpoint. Other nodes or services might use this URL to query the node's API.
-    ///
-    /// Default: `http://{rpc_bind}` (e.g., "http://0.0.0.0:20443" if rpc_bind is default).
+    /// ---
+    /// @default: `http://{rpc_bind}` (e.g., "http://0.0.0.0:20443" if [`NodeConfig::rpc_bind`] is default).
     pub data_url: String,
     /// The publicly accessible IPv4 address and port that this node advertises to peers for P2P connections.
-    /// This might differ from `p2p_bind` if the node is behind NAT or a proxy.
-    /// Note: The default value derivation might be unexpected, potentially using the `rpc_bind` address; explicit configuration is recommended if needed.
-    ///
-    /// Default: Derived from `rpc_bind` (e.g., "0.0.0.0:20443" if rpc_bind is default).
+    /// This might differ from [`NodeConfig::p2p_bind`] if the node is behind NAT or a proxy.
+    /// ---
+    /// @default: Derived from [`NodeConfig::rpc_bind`] (e.g., "0.0.0.0:20443" if [`NodeConfig::rpc_bind`] is default).
+    /// @notes:
+    ///   - The default value derivation might be unexpected, potentially using the
+    ///     [`NodeConfig::rpc_bind`] address; explicit configuration is recommended if needed.
     pub p2p_address: String,
     /// The private key seed, provided as a hex string in the config file, used specifically for the
     /// node's identity and message signing within the P2P networking layer.
     /// This is separate from the main [`NodeConfig::seed`].
-    ///
-    /// Default: Randomly generated 32 bytes.
+    /// ---
+    /// @default: Randomly generated 32 bytes
     pub local_peer_seed: Vec<u8>,
     /// A list of initial peer nodes used to bootstrap connections into the Stacks P2P network.
     /// Peers are specified in a configuration file as comma-separated strings in the
     /// format `"PUBKEY@IP:PORT"` or `"PUBKEY@HOSTNAME:PORT"`. DNS hostnames are resolved
     /// during configuration loading.
-    ///
-    /// Default: Empty vector `[]`.
+    /// ---
+    /// @default: `[]` (empty vector)
```

Comment:
similar comment as above regarding defining the unit in the `@default` section. 
i think globally we should either always include the unit type in `default`, or always defer to have it in `@notes`. 
my preference is for the latter, with the downside meaning there will be more text/formatting required but i think it would be easier to read and more importantly easier to parse. 

and after i wrote the above - perhaps a new key should be used?
ex:
```
@defaults: `[]`
@unit: empty vector
@notes:
  - this is a note
@toml: |
  this is a toml example
```